editor/src/components/canvas/ui-jsx-canvas-renderer/ui-jsx-canvas-component-renderer.tsx

```diff
@@ -380,20 +380,40 @@ function isRenderProp(prop: any): prop is { props: { [UTOPIA_PATH_KEY]: string }
 }
 
 function isElementInChildrenOrPropsTree(elementPath: string, props: any): boolean {
-  const childrenArr = React.Children.toArray(props.children).filter(React.isValidElement)
-  const elementIsChild = childrenArr.some((c) => (c.props as any)[UTOPIA_PATH_KEY] === elementPath)
-  if (elementIsChild) {
-    return true
+  const childrenArr = fastReactChildrenToArray(props.children)
+  for (let c of childrenArr) {
+    if ((c.props as any)[UTOPIA_PATH_KEY] === elementPath) {
+      return true
+    }
+
+    for (let p in props) {
+      if (React.isValidElement(p) && (p.props as any)[UTOPIA_PATH_KEY] === elementPath) {
+        return true
+      }
+    }
```

Comment:
This `for` loop should be outside of the containing `for` loop, and likewise for the `for (let p in props)` below